### DISCUSSION TITLE
Showoff: MacOS Statusbar Space Indicator

### DISCUSSION THREAD

**Discussion by xiamaz** [asmvik/yabai#1111] [Show and tell]
I have built a statusbar space indicator since I don't like custom statusbars (often very inefficient and one needs the normal macos statusbar often anyways. The icons can also be clicked for switching desktops. Currently this only uses yabai for switching desktops, but I am considering adding some more yabai status information.

The code can be found here: https://github.com/xiamaz/YabaiIndicator

<img width="1279" alt="Screen Shot 2021-12-28 at 01 10 27" src="https://user-images.githubusercontent.com/3340720/147514898-a14d8d2f-39ab-4081-89c6-90c5a63ec2e6.png">
)

**Comment by 24unix** [comment]
Nice idea. However, I built my own status bar a I like a big title bar of the actual browser tab.
Does it handle multi head setups?

**Reply by xiamaz** [reply]
Yes it does handle multi-head setups. In the screenshot two displays are active (the unfocused space is greyed out)

**Comment by 24unix** [comment]
Do you keep desktops in sync?

But I don't want to spam your thread, I’ll describe my problem in my own, but would be glad of feedback/ideas.

Maybe you might have a look: https://github.com/koekeishiya/yabai/discussions/759#discussioncomment-1879161

~~Which language did you chose for you statusbar?~~

Ignore the last question, I found you repo.

**Reply by xiamaz** [reply]
No, I don't keep them in sync. But I guess activating the correct spaces should do the trick? At least on mine the cursor jumps between displays on space focus, but that might just be something in my current yabai config.

**Comment by FelixKratz** [comment]
https://github.com/dshnkao/SpaceId

**Reply by xiamaz** [reply]
Somehow I also only managed to find it after beginning to write my own version. At least my version can be used to switch spaces too, so there's yet another solution to the same problem.

**Reply by Chandru-B** [reply]
@xiamaz Nice one!
Do you know how to get it installed?
I used to use https://github.com/cmacrae/spacebar but it is broken in release 4 of Yabai

**Reply by xiamaz** [reply]
Thanks. You can build it yourself in XCode, but I have also added a pre-release here: https://github.com/xiamaz/YabaiIndicator/releases/tag/0.1.0

**Comment by xiamaz** [comment]
Relying on Yabai Window information I have also managed to get proper window layout into desktops, but performance seems to be rather bad right now:

![image](https://user-images.githubusercontent.com/3340720/147894058-73d883e3-72f1-45c2-a1a1-b7d90a9107c8.png)

**Reply by ysomad** [reply]
Hello! Are you going to maintain Yabai Indicator?

**Reply by xiamaz** [reply]
I use it myself and might answer bug reports, but there currently are no plans for more features.

**Reply by ysomad** [reply]
Can you please provide some instructions how to install it? Currently using SpaceId, wanna try out YabaiIndicator.

**Comment by gldtn** [comment]
Awesome, I had just installed SpaceID but it seems like it's not really being maintained! Installed yours and it works great, can you create a brew package?
